### ISSUE TITLE
Add ray-casting based query for point lying inside/outside mesh

### ISSUE THREAD

**Issue by Shane-J-Latham** [AppliedMathematicsANU/pyemblite#4]
Would be good for the API to support multiple mesh objects in one scene, i.e. return inside/outside result for each mesh object in a scene.

**Comment by odf** [comment]
Agreed! That's already possible with the combination of `EmbreeScene` and `TriangleMesh`, but in that case the scene object does not retain the raw mesh data. For `EmbreeSceneExtended`, it should be fairly easy to maintain a list or dictionary of meshes stored in the instance and then in `intersections()` use `ray_hit.hit.geomID` to pick the correct one.
